### DISCUSSION TITLE
Simplify HA Controls Set

### DISCUSSION THREAD

**Discussion by fhteagle** [firstof9/openevse#195] [Ideas]
Now that GUI V2 and associated HTTP API changes are moving away from "sleeping" and "manual override", I think the legacy 2 switches and 2 sensors way of manual user control are ready to be obsoleted. Yes, set_override and clear_override are available to control from automation, and work just fine. However, for user-facing controls, I think there is a more elegant way to present the current status, and to allow for selection of OpenEVSE basic operations.

Toward that end, I created an input_select helper with the following options, and have been using that as the "sole control" for about 2 weeks now. The select options include:

- No HA Control       (i.e. "clear override")
- 0A                               (i.e. override set, state: disable, what used to be called sleeping)
- 6A                              (i.e. override set, state: active, charge_current: 6, auto-release: false)
- 7A 
...
- 16A                            (i.e. override set, state: active, charge_current at max_current_soft value from /config endpoint, auto-release: false)

Of course it took a pair of automations to send any changes to this input helper to the unit, and to update the select option if the unit was changed by an entity other than HA. 

However, it works very very well to present a single, space efficient, easy to read control/indicator on dashboards, etc. 

Therefore, I suggest evolving the select.*evse_max_current dropdown into what I have outlined above.

**Comment by firstof9** [comment]
Due to also supporting older firmware, specifically the 2.x versions, I cannot remove the switches.

**Reply by fhteagle** [reply]
Hmm. I had not thought about the need to support legacy hw / sw versions. 

Would it be possible to only expose the switches if the HA integration determines it is talking to an "old enough" unit, whatever that may be? I know this is a bit messy for the coding and dev process, but it is worth asking the question at least.

**Reply by firstof9** [reply]
Also technically "clear override" makes more sense as a `button` entity like the "Reset WiFi" button.

You are free to customize your template entities however you like however, but removing user choice isn't something I'd really like to do.
